schemas/com.snowplowanalytics.badrows/schema_violations/jsonschema/1-0-0

```diff
@@ -0,0 +1,167 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for a bad row resulting from schema violations",
+  "self": {
+    "vendor": "com.snowplowanalytics.badrows",
+    "name": "schema_violations",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+  "type": "object",
+  "properties": {
+    "failure": {
+      "type": "object",
+      "description": "Information regarding the schema violations",
+      "properties": {
+        "timestamp": {
+          "type": "string",
+          "description": "Timestamp at which the failure occurred",
+          "format": "date-time"
+        },
+        "messages": {
+          "type": "array",
+          "description": "List of failure messages associated with the schema violations",
+          "items": {
+            "type": "object",
+            "description": "Individual schema violation",
+            "anyOf": [
+              {
```

Comment:
I'm trying to figure out what's going on here. (Again - apologies for the dumb questions.) Can I confirm I've understood it right:

* This is an array with one item for each violation / issue that emerged validating the schemas for a particular event
* Each array can be one of three types, where different data points are available for each type of error:
  - For the first type four data points are captured (`schemaKey`, `instance`, `igluRepositories` and `message`)
  - For the second type we have the most detailed data set with the following fields: `schemaKey`, `instance`, `igluRepositories`, `property`, `value`, `expected`)
  - For the third we only have a single data point, `etc`
* Presumably the first two types correspond to two types of violation, where for the second type the field name and value that caused the violation have been parsed out of the self describing JSON, but for the first type they have not
* And the third type is a catchall for failures where the output doesn't conform to the above to types? 

If that is correct (and it might well not be), some questions:

1. What exactly is `instance`? It is a jsonpath to the field in question? In the self describing JSON? Or the schema? Can we name this so it is less ambiguous?
2. How comes the first type of error has a `message` property but the second type doesn't?
3. Could we do something to make it easier to identify which of the three types each object in the array is (e.g. add an additional enum field to all three types) to make it easier to query the array. (So an analyst can write a query that says, if the item in the array is of the second type, fetch the field value...)
4. I find the name of the single field in the third item type, `etc`, a bit confusing. Is there some context or a convention there that I'm not aware of that makes this name much more self-explanatory than I realise?

Reply:
Here's my attempt to formalize an error coming from Iglu client: https://gist.github.com/chuwy/f03247b2c399d134d428aa81863bdfc4, as the `schema_violation` is pretty much owned by Iglu client. 

Few moments to bear in mind:

* It has no Iglu URI (or `schemaKey`), which is however super-critical in a bad row itself. I belive it should be included in a top-level property defined by bad row and not returned by Iglu client (as it generally doesn't make sense for callee (client) to return the argument that already was passed to it)
* We don't have *much* control over `Invalid data` format, which is most important part of that schema.